From 9d373bd07b685c7c0e122cc1b5910400b444c1e5 Mon Sep 17 00:00:00 2001
From: NeKit <nekit1000@gmail.com>
Date: Sun, 5 Jul 2020 13:56:04 +0200
Subject: [PATCH] (halium) use 127.0.0.53 as fallback DNS server

Run by systemd-resolved in Ubuntu Touch. Needed to fix some
Android-side daemons requiring internet access like vendor AGPS.

Change-Id: I3c8dd8cb4d64d7632a853b12ff8c2f066534fcad
---
 libc/dns/resolv/res_init.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libc/dns/resolv/res_init.c b/libc/dns/resolv/res_init.c
index 2fb2a3d..be8b89d 100644
--- a/libc/dns/resolv/res_init.c
+++ b/libc/dns/resolv/res_init.c
@@ -200,8 +200,12 @@ __res_vinit(res_state statp, int preinit) {
 	}
 
 	memset(u, 0, sizeof(u));
+// Halium - systemd-resolved is listening at 127.0.0.53
+#define USELOOPBACK
 #ifdef USELOOPBACK
 	u[nserv].sin.sin_addr = inet_makeaddr(IN_LOOPBACKNET, 1);
+	// Ubuntu Touch specific - provided by systemd-resolved
+	u[nserv].sin.sin_addr.s_addr = inet_addr("127.0.0.53");
 #else
 	u[nserv].sin.sin_addr.s_addr = INADDR_ANY;
 #endif
-- 
2.45.2

